### ISSUE TITLE
Document use with Erlang

### ISSUE THREAD

**Issue by lpil** [hansihe/rustler#127]
Hello!

Thanks for this lib.

There is great documentation and examples for use in an Elixir project, but none for an Erlang project. Would it be possible to add this?

Thanks,
Louis

**Comment by scrogson** [comment]
@lpil currently, the `rustler_mix` project is a `mix` compiler. There is [this issue](https://github.com/hansihe/rustler/issues/85) which asks about `rebar3` support.

I think we should definitely provide some guidance about that but haven't taken the time to explore it yet.

**Comment by OvermindDL1** [comment]
The elixir -> erlang code should be a direct conversion, what issues are there?

**Comment by scrogson** [comment]
@OvermindDL1 I think the main thing is people who are building a project exclusively with `erlang` may not want to bring `elixir` into their project just to compile a `rust` crate.

**Comment by lpil** [comment]
Yes, I'd like to avoid adding Elixir to this project :)

**Comment by DeadZen** [comment]
Any examples yet?

**Comment by progamer71** [comment]
My reason is I prefer erlang syntax over elixir syntax.

**Comment by OvermindDL1** [comment]
> My reason is I prefer erlang syntax over elixir syntax.

Yeah many of us do, Erlang's syntax is just so superior, but Elixir's Macro's and everything they bring in is the main thing that keeps me working in Elixir.  ^.^;

**Comment by russelldb** [comment]
there's a good blog post here https://medium.com/@vonmo/erlang-safe-optimization-with-nif-on-rust-6000b0e9e4bf that covers the basics of running rustler with erlang. I use rustler with erlang so that I can use quviq EQC to test Rust components. I have a few issues (I don't know how to have multiple modules in the same crate for example) but in general it works really well. 

I don't have the time right now, but when I do I can write up how to use rustler from erlang

**Comment by rickpayne** [comment]
I started work on using your rustler_tests from a rebar3 project at https://github.com/rickpayne/rustler_test
